tests/BasicErasedProvisionTests.fs

```diff
@@ -283,289 +127,27 @@ let nonPrimitives =
-    let tp = TypeProviderForNamespaces(cfg)
-    let mscorlib31 = match tp.TargetContext.TryBindSimpleAssemblyNameToTarget("mscorlib") with Choice1Of2 asm -> asm | Choice2Of2 err -> failwithf "couldn't bind mscorlib, err: %O" err
-    // primitive types with element types are ALWAYS equivalent the design-time types
-    for tname, sourceType, _ in primitives do
-        let targetType = mscorlib31.GetType(tname)
-        Assert.NotEqual(targetType, sourceType)
-    // System.Void is given a special treatment
-    let targetType = mscorlib31.GetType("System.Void")
-    Assert.NotEqual(targetType, typeof<Void>)
-
-[<Fact>]
-let ``Check target non-primitive types are different to design-time types``() : unit  = 
-    let refs = Targets.DotNet45FSharp31Refs()
-    let cfg = Testing.MakeSimulatedTypeProviderConfig (__SOURCE_DIRECTORY__, refs.[0], refs)
-    let tp = TypeProviderForNamespaces(cfg)
-    let mscorlib31 = match tp.TargetContext.TryBindSimpleAssemblyNameToTarget("mscorlib") with Choice1Of2 asm -> asm | Choice2Of2 err -> failwithf "couldn't bind mscorlib, err: %O" err
-    // non-primitive types should be _not_ be equal - we should see the target type in the referenced assemblies
-    for tname, sourceType, _ in nonPrimitives do
-        let targetType = mscorlib31.GetType(tname)
-        Assert.NotEqual(targetType, sourceType)
-
-
-[<Fact>]
-let ``Check target enum types gives right values``() : unit  = 
-    let refs = Targets.DotNet45FSharp31Refs()
-    let cfg = Testing.MakeSimulatedTypeProviderConfig (__SOURCE_DIRECTORY__, refs.[0], refs)
-    let tp = TypeProviderForNamespaces(cfg)
-    let dayOfWeekType = typeof<System.DayOfWeek>
-    let dayOfWeekTypeT = tp.TargetContext.ConvertSourceTypeToTarget dayOfWeekType
-    printfn "Enums #1"
-    Assert.True(dayOfWeekType.IsEnum)
-    printfn "Enums #2"
-    Assert.Equal(dayOfWeekType.GetEnumUnderlyingType().FullName, "System.Int32")
-    printfn "Enums #3"
-    Assert.True(dayOfWeekTypeT.IsEnum)
-    printfn "Enums #4"
-    Assert.Equal(dayOfWeekTypeT.GetEnumUnderlyingType().FullName, "System.Int32")
-    printfn "Done Enums"
-
-[<Fact>]
-let ``Check target delegate types gives right values``() : unit  = 
-    let refs = Targets.DotNet45FSharp31Refs()
-    let cfg = Testing.MakeSimulatedTypeProviderConfig (__SOURCE_DIRECTORY__, refs.[0], refs)
-    let tp = TypeProviderForNamespaces(cfg)
-    for delegateType in [ typeof<Func<int,int>>; typeof<System.Converter<int,int>>; typeof<System.Action> ] do
-        let delegateTypeT = tp.TargetContext.ConvertSourceTypeToTarget delegateType
-        printfn "Delegates #1, delegateType = %A" delegateType
-        Assert.True(delegateType.IsSubclassOf(typeof<System.Delegate>))
-        printfn "Delegates #2, delegateType = %A" delegateType
-        Assert.True(delegateTypeT.IsSubclassOf(typeof<System.Delegate>))
-
-[<Fact>]
-let ``Check type remapping functions work for primitives``() : unit  = 
-    let refs = Targets.DotNet45FSharp31Refs()
```

Comment:
There is a huge amount of important testing being deleted here.  

Type providers still have to correctly cross-target (they can be used in netstandard2.0, netstandard2.1, netcoreapp3.0, net5 projects etc. etc) - this doesn't change just because we assume TPRTC are netstandard 2.0/2.1.  So all this testing of corner cases in cross-targeting is still relevant.

Reply:
These aren't necessary if we wish to make TPs a .NET Standard 2.0-only concept going forward.

Reply:
My understanding of these tests is that they're ultimately just doing an ad-hoc implementation of what .NET Standard 2.0 assembly forwarding already does for us

Reply:
No as I understand things that's not correct.

.NET Standard 2.0 TPDTC still have to multi-target, that is, they may be used in a project whose target reference assemblies are .NET Standard 2.0, .NET Standard 2.1, .NET CoreApp 3.0 or whatever else.   They still have to generate code correctly for the correct set of target reference assemblies.  This includes generative assemblies.  So while TPDTC are .NET Standard 2.0, they are in fact code generators for a whole range of target frameworks depending on the reference assembly set they are given. That's what this code is testing, and what type remapping is for.  

To put it another way - there's nothing about the simplification of the engineering assumptions in this repo that removes the requirement that TPDTC still have to implement type remapping.